File: benchmarks/matrix/matrix.js
const N = 150

const A = []
const B = []
const C = []

for (let i = 0; i < N; i++) {
	A.push([])
	B.push([])
	C.push([])

	for (let j = 0; j < N; j++) {
		A[i].push((i + j) % 100)
		B[i].push((i * j) % 100)
		C[i].push(0)
	}
}

for (let i = 0; i < N; i++) {
	for (let j = 0; j < N; j++) {
		let sum = 0

		for (let k = 0; k < N; k++) {
			sum += A[i][k] * B[k][j]
		}

		C[i][j] = sum
	}
}

console.log(C[0][0])
console.log(C[N - 1][N - 1])

File: benchmarks/matrix/matrix.pl
use strict;
use warnings;

my $N = 150;

my @A;
my @B;
my @C;

for my $i (0 .. $N - 1) {
    $A[$i] = [];
    $B[$i] = [];
    $C[$i] = [];

    for my $j (0 .. $N - 1) {
        push @{$A[$i]}, (($i + $j) % 100);
        push @{$B[$i]}, (($i * $j) % 100);
        push @{$C[$i]}, 0;
    }
}

for my $i (0 .. $N - 1) {
    for my $j (0 .. $N - 1) {
        my $sum = 0;

        for my $k (0 .. $N - 1) {
            $sum += $A[$i][$k] * $B[$k][$j];
        }

        $C[$i][$j] = $sum;
    }
}

print "$C[0][0]\n";
print "$C[$N-1][$N-1]\n";